### PR TITLE
Add a `client_scope` field to `ClientConfig`

### PR DESCRIPTION
**Pull request by eugeneo**
Commit Message:
Add a `client_scope` field to `ClientConfig`

Additional Description:
Extends an CSDS API with an additional field that includes extra information for implementations that can have multiple unrelated configs.

Risk Level:
Low

Testing:
Manual

Docs Changes:
None

Release Notes:
None

Platform Specific Features:
None

### CODE REVIEW

**Comment by markdroth** [comment]
For context on why this field is needed, see https://github.com/grpc/proposal/blob/master/A71-xds-fallback.md.

File: api/envoy/service/status/v3/csds.proto
Comment (markdroth) [inline api/envoy/service/status/v3/csds.proto:192]: I think the field id should be 4, not 10.

**Review by lizan — APPROVED** [review]

**Comment by lizan** [comment]
can you fix DCO and CI?

**Comment by eugeneo** [comment]
> can you fix DCO and CI?

Fixed DCO. `Mobile/Core` failure seems to have to do with infra:
```
Suppressed: java.io.IOException: Error while uploading artifact with digest '9805f9f96ce4636319e3cb973bfab92899012dc903e26b2a85a599bec3[542](https://github.com/envoyproxy/envoy/actions/runs/7586833103/job/20665938490#step:8:556)a57/243'
		at com.google.devtools.build.lib.remote.ByteStreamUploader.lambda$uploadBlobAsync$0(ByteStreamUploader.java:171)
		at com.google.common.util.concurrent.AbstractCatchingFuture$AsyncCatchingFuture.doFallback(AbstractCatchingFuture.java:203)
		at com.google.common.util.concurrent.AbstractCatchingFuture$AsyncCatchingFuture.doFallback(AbstractCatchingFuture.java:190)
		at com.google.common.util.concurrent.AbstractCatchingFuture.run(AbstractCatchingFuture.java:133)
		... 57 more
	Caused by: io.grpc.StatusRuntimeException: RESOURCE_EXHAUSTED: No available CAS nodes
		at io.grpc.Status.asRuntimeException(Status.java:535)
		... 10 more
```

**Comment by markdroth** [comment]
Can one of the Envoy maintainers take a look at the remaining CI failure and advise?  We'd like to get this merged to unblock some downstream work.  Thanks!

**Comment by wbpcode** [comment]
cc @phlax Seems this is a CI self's issue?

**Comment by phlax** [comment]
> Seems this is a CI self's issue?

yep - its engflow RBE running out of resources cc @BillyAutrey

**Comment by adisuissa** [comment]
> cc @phlax Seems this is a CI self's issue?

Should this PR be forced-merged?

**Comment by phlax** [comment]
from CI pov i think yes

**Pull request merged**
